prov/verbs/src/windows/verbs_nd_ov.c

```diff
@@ -306,85 +321,72 @@ void nd_cm_connect_ack(struct nd_event_base *base, DWORD bytes)
 			event->param.private_data_len;
 		cm_event->event.param.conn.private_data =
 			event->param.private_data;
-		// Intentional fall through.
+		insert_cm_event(event->channel, cm_event);
+		nd_notify_disconnect(id, connector);
+		break;
 
 	case ND_CM_ACCEPT:
-		// If we fell through from above, then we need to allocate the cm_event.
-		if (!cm_event)
-			cm_event = nd_allocate_cm_event(
-				id, RDMA_CM_EVENT_ESTABLISHED);
-
-		id_nd = container_of(id, struct nd_cm_id, id);
-		id_nd->peer_event.state = ND_CM_DISCONNECTED;
-
-		hr = connector->lpVtbl->NotifyDisconnect(
-			connector, &id_nd->peer_event.base.ov);
-		FI_LOG(&vrb_prov, FAILED(hr) ? FI_LOG_WARN : FI_LOG_DEBUG,
-		       FI_LOG_EP_CTRL,
-		       "IND2Connector::NotifyDisconnect: hr=0x%08lx; ov=%p\n",
-		       hr, &id_nd->peer_event.base.ov);
+		cm_event = nd_allocate_cm_event(id, RDMA_CM_EVENT_ESTABLISHED);
+		insert_cm_event(event->channel, cm_event);
+		nd_notify_disconnect(id, connector);
 		break;
 
+	case ND_CM_DISCONNECT:
 	case ND_CM_DISCONNECTED:
 		cm_event = nd_allocate_cm_event(id, RDMA_CM_EVENT_DISCONNECTED);
+		insert_cm_event(event->channel, cm_event);
 		break;
 
 	default:
 		break;
 	}
-
-	dlistfd_insert_tail(&cm_event->entry, &ch_nd->q);
-	VRB_DBG(FI_LOG_EQ, "EQ SET event:%p %p %d\n", cm_event->event.id,
-		cm_event->event.listen_id, cm_event->event.event);
 }
 
 void nd_cm_connect_nack(struct nd_event_base *base, DWORD bytes, DWORD error)
 {
 	struct nd_cm_connect_event *event;
-	struct nd_cm_event *entry_nd;
+	struct nd_cm_event *cm_event = NULL;
 	HRESULT hr;
-	struct nd_event_channel *ch_nd;
 
 	VRB_TRACE(FI_LOG_FABRIC, "\n");
 
 	event = container_of(base, struct nd_cm_connect_event, base);
 
 	if (error == ND_CANCELED) {
 		return;
-	} else if ((event->state == ND_CM_DISCONNECTED) &&
+	} else if ((event->type == ND_CM_DISCONNECTED) &&
```

Comment:
nit: the if () statement returns, so the else can be dropped here.  This is just a new if statement.